resolving/git-resolver/src/parseBareSpecifier.ts

```diff
@@ -169,16 +169,21 @@ function parseGitParams (committish: string | null): GitParsedParams {
 // handle SCP-like URLs
 // see https://github.com/yarnpkg/yarn/blob/5682d55/src/util/git.js#L103
 function correctUrl (gitUrl: string): string {
-  const parsed = urlLib.parse(gitUrl.replace(/^git\+/, '')) // eslint-disable-line n/no-deprecated-api
-
-  if (parsed.protocol === 'ssh:' &&
-    parsed.hostname &&
-    parsed.pathname &&
-    parsed.pathname.startsWith('/:') &&
-    parsed.port === null) {
-    parsed.pathname = parsed.pathname.replace(/^\/:/, '')
-    return urlLib.format(parsed)
+  let _gitUrl = gitUrl.replace(/^git\+/, '')
+  if (_gitUrl.startsWith('ssh://')) {
+    const hashIndex = _gitUrl.indexOf('#')
+    let hash = ''
+    if (hashIndex !== -1) {
+      hash = _gitUrl.slice(hashIndex)
+      _gitUrl = _gitUrl.slice(0, hashIndex)
+    }
+    const [auth, pathname] = _gitUrl.slice(6).split('/')
+    const [, host] = auth.split('@')
+    if (host.includes(':') && !/:\d+$/.test(host)) {
```

Comment:
1. This needs a comment, IMO. The `6` looks like a magic number, and there's no clear sign that discarding all of the following "/"-components is intentional
2. What if there are multiple "@"s?
3. There's no guarantee that the ":" is at the same index, for example `"localhost:40:40"`

Reply:
> 1. This needs a comment, IMO. The `6` looks like a magic number, 

I think the judgment in line 173 should be enough to explain the situation.
> and there's no clear sign that discarding all of the following "/"-components is intentional

I don't understand what you mean by this part.

Reply:
> I think the judgment in line 173 should be enough to explain the situation.

Fair enough. In that case, I'd write something like:
```ts
/** SSH URI Scheme */
const PREFIX = 'ssh://', PREFIX_LEN = PREFIX.length
if (_gitUrl.startsWith(PREFIX)) {
// (snip)
  const [auth, pathname] = _gitUrl.slice(PREFIX_LEN).split('/')
```

> > and there's no clear sign that discarding all of the following "/"-components is intentional
> 
> I don't understand what you mean by this part.

`pathname` should be named `endpoint`:
```js
'example.org/api/main'.split('/')[1] === 'api'
```
If it was a full path-name, then it should include everything after the 1st slash. But I suspect I'm missing something :thinking:

Reply:
Thank you. We seem to be missing relevant test cases, which leads to a bug. It has been fixed in #9548.